### ISSUE TITLE
Changing tab (active window) while animation

### ISSUE THREAD

**Issue by l2aelba** [legomushroom/mojs#58]
I have a problem when user changing window tab to another page. So the animation will not be complete and few bug (like not all element beging drawn when user going back to the current tab)

**And I fixed by this :**

``` javascript
document.addEventListener('visibilitychange', function _func() {
	if (!document.hidden && !timeline._isCompleted) {
		timeline.replay();
	}
	if (timeline._isCompleted) {
		document.removeEventListener('visibilitychange', _func);
	}
}, false);
```

But I want to know, **any trick or better solution can I do ?** _(or im missing something?)_

Thanks

**Comment by legomushroom** [comment]
hi @l2aelba! Thanks for the issue!

Never encountered the issue before, but the problem is indeed there and the culprit is `requestAnimationFrame` that don't fire on inactive tab thus ruins the animation. Your trick might work, but to understand your intention better - how do you want(or expect) it should work? Like for instance just pause the animation and then start it when user will return to the tab, or for instance, be in the perfectly completed state when user returns?

Thanks a lot for your feedback!

**Comment by l2aelba** [comment]
@legomushroom , This issue should be your choise :D , I just quick fixed this issue so long.

**Comment by legomushroom** [comment]
@l2aelba ok, will include the fix in the nearest release. Thanks a lot!

**Comment by legomushroom** [comment]
Hi, just released the `0.288.1` - it `pauses` and `resumes` all animation on `visibilityChange` automatically 🎉 ! Thanks for your feedback!

**Comment by l2aelba** [comment]
Good job @legomushroom 👍
